### ISSUE TITLE
task01: professional workflow

### ISSUE THREAD

**Issue by Revilllo7** [Revilllo7/computational-intelligence-class#61]
Fork the repo and do the following:

a) draw the project architecture
Draw the block scheme of the project in `draw.io` or something else. Write a note about the process and the flow
`fetch` -> `preprocess` -> `train` -> `evaluate` -> `visualise` -> `compare`

Write which directories are used for input and output.

Aim: understand `configs/`, `src/cli/`, `src/experiments`, `data/`, `models/`, `reports/`, `tests/`

b) run the project in the terminal
run `make setup`, `make pipeline`, `make tests`, `make pyright`, `make ruff`. Then explain what each command does.

Aim: getting accustomed with uv, makefile, tests, lint and type-checking.

c) read the experiment configuration
Read the .yaml files in `configs/` and compare them. Write what are the differences for the experiments and which configuration fields impacted the model, pre-processing and artifact save paths.

Aim: understanding how a project is steered by the configuration files instead of re-writing code.

d) follow an experiment run
Run one experiment and follow it along. Check what's happening in `src/cli/run_experiment.py` up to the point of saving the reports in `reports/` and `models/`. Write a short note about what functions are called and in what order.

Aim: Seeing how CLI works and how the code is separated into layers.

e) explain the base class of the experiment
Read `src/experiments/base.py` and write what elements are common for all experiments and which ones are interchangeable

Aim: understanding the idea of base class, shared pipeline and project expansion through new experiments.

f) check the config validation
Read `src/utils/config.py`. Then write two FAULTY .yaml files which should get rejected by the validation and explain why.

Aim: understanding the role of Pydantic and the reason for config validation before running an experiment.

g) add a new experiment variant without changing the code
Copy one of the existing config files and create a new experiment by changing arguments like `epochs`, `learning_rate` or `hidden_dims`. Run it and compare the results with the base experiment.

Aim: learning how to expand a project through config.

h) add new preprocessing
Expand the project by adding a new preprocessing strategy (example: `robust`). Add necessary code services

Aim: refining code modification alongside tests

i) add a new model or new experiment type
Add a simple model variant in regards to classification and register it so it's runnable through config. Make a separate .yaml file.

Aim: Seeing how the experiment registry works and how the architecture is expandable without breaking existing code.

j) expand the result reports
Add a new metric or plot. Make sure the result saves into `reports/` and is comparable with other results.

Aim: understanding that the ML projects doesn't end at just a model, but also needs a proper report.
 
k) add error tests
Add a test that checks what happens when user inputs wrong experiment name or wrong config.

Aim: learning how to write tests not only for correct cases, but also errors and edge cases

l) add your own CLI command
Add a new CLI command for (example: report summary or listing available configs).

Aim: practicing expansion of terminal interface for the project.

m) change the dataset for `diagnosis.csv`
Change the iris dataset from previous labs for the `diagnosis.csv` file.
Write which elements of the project had to be changed and which were unchanged.

Aim: checking if the project architecture correctly separates the general elements from data specific elements
